selfdrive/car/hyundai/values.py

```diff
@@ -1325,9 +1325,18 @@ class Buttons:
   CAR.KIA_NIRO_EV_2ND_GEN: {
     (Ecu.fwdRadar, 0x7d0, None): [
       b'\xf1\x00SG2_ RDR -----      1.00 1.01 99110-AT000         ',
+      b'\xf1\x8799110AT000\xf1\x00SG2_ RDR -----      1.00 1.01 99110-AT000         ',
     ],
     (Ecu.fwdCamera, 0x7c4, None): [
       b'\xf1\x00SG2EMFC  AT EUR LHD 1.01 1.09 99211-AT000 220801',
+      b'\xf1\x00SG2EMFC  AT USA LHD 1.01 1.09 99211-AT000 220801',
+    ],
+    (Ecu.eps, 0x7d4, None): [
+      b'\xf1\x00SG2 MDPS C A.00 1.03 56310AO000  4SGVC103',
+      b'\xf1\x8756310AO000 \xf1\x00SG2 MDPS C A.00 1.03 56310AO000  4SGVC103',
+    ],
+    (Ecu.cornerRadar, 0x7b7, None): [
+      b'\xf1\x00SG2 BCW RR 1.01 , 1.04 (wQ\x83"\x02\x16\x00c',
```

Comment:
As you can see from the failing unit test, we do not allow non-powertrain-bus ECUs in the fingerprints for CAN FD cars since they are behind the OBD2 gateway, which manages communication and makes it spotty.

Reply:
Will test without `Ecu.eps` & `Ecu.cornerRadar`. Thanks for the nudge in the right direction 👍

Reply:
Help, kia niro HYBRID 2023. I'm on 0.91 current version but says car unrecognized? The car support listing shows it should be supported but I can't seem to get it to work and can't figure out where/how to ask for support

Reply:
> Help, kia niro HYBRID 2023. I'm on 0.91 current version but says car unrecognized? The car support listing shows it should be supported but I can't seem to get it to work and can't figure out where/how to ask for support

Try this guide: https://github.com/commaai/openpilot/wiki/Fingerprinting
I found it to be helpful for me as I had the very same issue.